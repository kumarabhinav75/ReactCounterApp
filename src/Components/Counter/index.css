.button1 {
  background: yellow;
  color: purple;
}
.button2 {
  background: purple;
  color: yellow;
}