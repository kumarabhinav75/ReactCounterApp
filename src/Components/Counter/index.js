import React, { Component } from 'react';
import './index.css'
import Button from '../Button';

class Counter extends Component {

  state = {
    count: this.props.initialValue
  }

  incrementCount = () => this.setState({count: this.state.count+1});
  decrementCount = () => this.setState({count: this.state.count-1});

  render(){
    return (
      <div>
      <p>Counter : {this.state.count}</p>
      <Button className="button1" buttonText="Increase" onClick={this.incrementCount} />
      <Button className="button2" buttonText="Decrease" onClick={this.decrementCount} />
      </div>
    );
    }
}
export default Counter;