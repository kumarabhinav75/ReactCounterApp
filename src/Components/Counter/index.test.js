import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Counter from './index';

describe('Counter', () =>{
  it('should render a  button', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should test the initalValue', () => {
    const wrapper = shallow(<Counter initialValue = {10} />);
    expect(wrapper.instance().state.count).toEqual(10);
  });
  it('Should increment the state.count', () => {
    const wrapper = shallow(<Counter initialValue = {10} />);
    wrapper.instance().incrementCount();
    expect(wrapper.instance().state.count).toEqual(11);
  });
  it('Should decrement the state.count', () => {
    const wrapper = shallow(<Counter initialValue = {10} />);
    wrapper.instance().decrementCount();
    expect(wrapper.instance().state.count).toEqual(9);
  });
});