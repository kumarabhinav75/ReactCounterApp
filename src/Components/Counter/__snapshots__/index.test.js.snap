// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render a  button 1`] = `
<div>
  <p>
    Counter : 
  </p>
  <button
    className="button1"
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    className="button2"
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
